### PR TITLE
update to 0.50

### PR DESCRIPTION
**Pull request by tac0turtle**

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -1,8 +1,13 @@
 package app
 
 import (
+	"context"
+	"cosmossdk.io/x/tx/signing"
 	"encoding/json"
 	"fmt"
+	"github.com/cosmos/cosmos-sdk/codec/address"
+	"github.com/cosmos/cosmos-sdk/std"
+	"github.com/cosmos/gogoproto/proto"
```
Comment (ash-burnt) [inline app/app.go:10]: might be worth running our formatters for consistent import arrangement:
`make format`

File: app/export.go [hunk 2]
```diff
@@ -46,7 +52,7 @@ func (app *WasmApp) ExportAppStateAndValidators(forZeroHeight bool, jailAllowedA
 // prepare for fresh start at zero height
 // NOTE zero height genesis is a temporary feature which will be deprecated
 //
-//	in favour of export at a block height
+//	in favor of export at a block height
```
Comment (ash-burnt) [inline app/export.go:55]: 🇺🇸

File: app/export.go [hunk 3]
```diff
@@ -140,33 +172,41 @@ func (app *WasmApp) prepForZeroHeightGenesis(ctx sdk.Context, jailAllowedAddrs [
...
-	counter := int16(0)
+	iter := storetypes.KVStoreReversePrefixIterator(store, stakingtypes.ValidatorsKey)
 
 	for ; iter.Valid(); iter.Next() {
 		addr := sdk.ValAddress(stakingtypes.AddressFromValidatorsKey(iter.Key()))
-		validator, found := app.StakingKeeper.GetValidator(ctx, addr)
-		if !found {
+		validator, err := app.StakingKeeper.GetValidator(ctx, addr)
+		if err != nil {
 			panic("expected validator, not found")
```
Comment (ash-burnt) [inline app/export.go:210]: should prob log this error

File: x/mint/types/codec.go [hunk 4]
```diff
@@ -25,9 +22,9 @@ func init() {
 	// Register all Amino interfaces and concrete types on the authz  and gov Amino codec
 	// so that this can later be used to properly serialize MsgGrant and MsgExec
 	// instances.
-	RegisterLegacyAminoCodec(authzcodec.Amino)
-	RegisterLegacyAminoCodec(govcodec.Amino)
-	RegisterLegacyAminoCodec(groupcodec.Amino)
+	//RegisterLegacyAminoCodec(authzcodec.Amino)
+	//RegisterLegacyAminoCodec(govcodec.Amino)
+	//RegisterLegacyAminoCodec(groupcodec.Amino)
```
Comment (ash-burnt) [inline x/mint/types/codec.go:27]: Do amino codecs not require registration anymore?

File: x/xion/types/msgs.go [hunk 5]
```diff
@@ -96,7 +96,7 @@ func (msg MsgMultiSend) ValidateBasic() error {
 		return banktypes.ErrNoOutputs
 	}
 
-	return banktypes.ValidateInputsOutputs(msg.Inputs, msg.Outputs)
+	return banktypes.ValidateInputOutputs(msg.Inputs[0], msg.Outputs) // TODO change to only use one input
```
Comment (ash-burnt) [inline x/xion/types/msgs.go:99]: This was meant to model the existing Bank MultiSend, what does it do here?

File: app/app.go [hunk 6]
```diff
@@ -610,10 +598,14 @@ func NewWasmApp(
 	app.TokenFactoryKeeper = tokenfactorykeeper.NewKeeper(
 		appCodec,
 		keys[tokenfactorytypes.StoreKey],
+		maccPerms,
 		app.AccountKeeper,
 		app.BankKeeper,
 		app.DistrKeeper,
 		tokenFactoryCapabilities,
+		func(ctx context.Context, addr string) bool { // TODO decide what to do
```
Comment (ash-burnt) [inline app/app.go:606]: Is there any context around this call?

File: x/xion/keeper/msg_server.go [hunk 7]
```diff
@@ -124,7 +125,7 @@ func (k msgServer) MultiSend(goCtx context.Context, msg *types.MsgMultiSend) (*t
 		outputs = append(outputs, banktypes.NewOutput(feeCollectorAcc, totalPlatformCoins))
 	}
 
-	err := k.bankKeeper.InputOutputCoins(ctx, msg.Inputs, outputs)
+	err := k.bankKeeper.InputOutputCoins(ctx, msg.Inputs[0], outputs) // TODO decide on how to handle multiple inputs
```
Comment (ash-burnt) [inline x/xion/keeper/msg_server.go:128]: What does the existing Bank MultiSend do in this instance? This was basically a straight copy

File: app/ante.go [hunk 8]
```diff
@@ -57,8 +57,8 @@ func NewAnteHandler(options HandlerOptions) (sdk.AnteHandler, error) {
 	if options.GlobalFeeSubspace.Name() == "" {
 		return nil, errorsmod.Wrap(sdkerrors.ErrNotFound, "globalfee param store is required for AnteHandler")
 	}
-	if options.TXCounterStoreKey == nil {
-		return nil, errorsmod.Wrap(sdkerrors.ErrLogic, "tx counter key is required for ante builder")
+	if options.TXCounterStoreService == nil {
+		return nil, errorsmod.Wrap(sdkerrors.ErrLogic, "wasm store service is required for ante builder")
```
[the remark this answers is not in the document]
Comment (tac0turtle) [reply app/ante.go:61]: it is a separate store used by cosmwasm

File: app/export.go (hunk 3 above)
Comment (tac0turtle) [reply app/export.go:210]: added an error to the panic and made it more explicit

File: x/xion/types/msgs.go (hunk 5 above)
Comment (tac0turtle) [reply x/xion/types/msgs.go:99]: it recreates the same logic as before except the inputs is an array. the above code makes sure the length is always 1 making this check safe

File: x/xion/keeper/msg_server.go (hunk 7 above)
Comment (tac0turtle) [reply x/xion/keeper/msg_server.go:129]: the logic stayed the same, in the cosmos sdk we opted to remove the validatebasic function and instead do the checks in the msg execution phase. the approach that is done here is still safe and the logic stays the same.

**Review by jgimeno — APPROVED** [review]

File: integration_tests/go.mod [hunk 9]
```diff
@@ -48,38 +50,48 @@ require (
...
-	github.com/cometbft/cometbft-db v0.9.1 // indirect
+	github.com/cometbft/cometbft v0.38.7 // indirect
+	github.com/cometbft/cometbft-db v0.12.0 // indirect
 	github.com/confio/ics23/go v0.9.0 // indirect
 	github.com/cosmos/btcutil v1.0.5 // indirect
-	github.com/cosmos/cosmos-proto v1.0.0-beta.4 // indirect
+	github.com/cosmos/cosmos-proto v1.0.0-beta.5 // indirect
 	github.com/cosmos/go-bip39 v1.0.0 // indirect
 	github.com/cosmos/gogogateway v1.2.0 // indirect
+<<<<<<< HEAD
```
Comment (seantking) [inline integration_tests/go.mod:75]: Need to remove the git merge tags :)

**Pull request closed without merging**